modules/candidate_list/php/validateids.class.inc

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+* The candidate list menu
+*
+* PHP Version 5
+*
+* @category Main
+* @package  Candidate_List
+* @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+* @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+* @link     https://www.github.com/aces/Loris/
+*/
+namespace LORIS\candidate_list;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * Implements the candidate_list menu
+ *
+* @category Main
+* @package  Candidate_List
+* @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+* @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+* @link     https://www.github.com/aces/Loris/
+ */
+class ValidateIDs extends \NDB_Page
```

Comment:
design question:

is this really a `page` ?? what does it have in common with page components ? isnt it more like a GENERAL request than anything else (applies to all ajax scripts) especially that the first line says "skip template" kind of proving my point

the way I see it, ajax scripts in loris are either just a huge mess or one of 3 things:
- request for validation
- request for data (getData)
- request for saving data (fileUpload, addThis, editThat,....)

shouldn't we just have at least these 3 as interfaces and have newly created classes implement one or more of them ?

Reply:
NDB_Page is the root of our endpoint/page type object hierarchy at the moment. There's nowhere higher for it to go.

Reply:
thats too bad then i guess